File: frontend/src/services/reportService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/reports";

// Get all reports
export const getReports = () => axios.get(API_URL);

// Add report
export const addReport = (report) =>
  axios.post(API_URL, report);

// Update report
export const updateReport = (id, report) =>
  axios.put(`${API_URL}/${id}`, report);

// Delete report
export const deleteReport = (id) =>
  axios.delete(`${API_URL}/${id}`);

File: backend/src/main/java/com/firealarm/backend/service/ReportService.java
package com.firealarm.backend.service;

import com.firealarm.backend.entity.Report;
import com.firealarm.backend.repository.ReportRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ReportService {

    @Autowired
    private ReportRepository repository;

    // Get all reports
    public List<Report> getAllReports() {
        return repository.findAll();
    }

    // Add report
    public Report saveReport(Report report) {
        return repository.save(report);
    }

    // Update report
    public Report updateReport(Long id, Report report) {

        Report existing = repository.findById(id).orElseThrow();

        existing.setReportName(report.getReportName());
        existing.setReportType(report.getReportType());
        existing.setGeneratedDate(report.getGeneratedDate());
        existing.setStatus(report.getStatus());

        return repository.save(existing);
    }

    // Delete report
    public void deleteReport(Long id) {
        repository.deleteById(id);
    }
}